x/rollapp/keeper/msg_server_mark_vulnerable_rollapps.go

```diff
@@ -62,19 +62,23 @@ func (k Keeper) MarkVulnerableRollapps(ctx sdk.Context, drsVersions []string) (i
 			logger.With("rollapp_id", rollapp.RollappId).Info("no latest state info for rollapp")
 			continue
 		}
-		// TODO: this check may be deleted once empty DRS version is marked vulnerable
-		//  https://github.com/dymensionxyz/dymension/issues/1233
-		if info.DrsVersion == "" {
-			logger.With("rollapp_id", rollapp.RollappId).Info("no DRS version set for rollapp")
-		}
 
-		_, vulnerable := vulnerableVersions[info.DrsVersion]
-		if vulnerable {
-			err := k.MarkRollappAsVulnerable(ctx, rollapp.RollappId)
-			if err != nil {
-				return 0, fmt.Errorf("freeze rollapp: %w", err)
+		for _, bd := range info.BDs.BD {
```

Comment:
the iteration here can be costly and potentially lead to dos as we currently don't limit the num blocks in a batch (though it's limited by tx size but still can be very high).
I think we can simply check the first and last block, cause he chances of DRS version changing in between is highly unlikely and even somehow it did, it's not that interesting to fork the rollap for.

Reply:
we only check last